### PR TITLE
Improve live managed identity tests

### PR DESCRIPTION
**Pull request by chlowell**
Adding live managed identity tests which use Key Vault to verify credentials acquire valid tokens. These new tests will be skipped in normal runs unless a couple of unwieldy environment variables unlikely to be set accidentally have values. This also includes a Dockerfile so the new tests can be run in a container, allowing them to execute in AKS and ACI.

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/identity/azure-identity/tests/helpers.py [hunk 1]
```diff
@@ -59,53 +58,73 @@ def build_aad_response(  # simulate a response from AAD
...
+        if self.base_url and self.base_url != request.url.split("?")[0]:
+            discrepancies.append("url:\n\texpected base url '{}', actual was '{}'".format(self.base_url, request.url))
+
+        if self.url and self.url != request.url:
+            add_discrepancy("url", self.url, request.url)
+
+        if self.url_substring and self.url_substring not in request.url:
+            discrepancies.append("url: '{}' not in '{}'".format(self.url_substring, request.url))
+
+        if self.authority and url.netloc != self.authority:
```
Comment (KieranBrantnerMagee) [inline sdk/identity/azure-identity/tests/helpers.py]: url -> self.url or request.url? (and below)

Comment (chlowell) [reply sdk/identity/azure-identity/tests/helpers.py]: This is correct, if confusing: `url` is `request.url`, parsed; if we were given an expected authority, the actual URL's netloc should match it. `url` is only used here, so I renamed it and moved its declaration closer.

File: sdk/identity/azure-identity/tests/helpers.py [hunk 2]
```diff
@@ -59,53 +58,73 @@ def build_aad_response(  # simulate a response from AAD
...
-            assert self.url_substring in request.url
-        if self.method:
-            assert request.method == self.method
+        discrepancies = []
+
+        def add_discrepancy(name, expected, actual):
+            discrepancies.append("{}:\n\t expected: {}\n\t   actual: {}".format(name, expected, actual))
+
+        if self.base_url and self.base_url != request.url.split("?")[0]:
+            discrepancies.append("url:\n\texpected base url '{}', actual was '{}'".format(self.base_url, request.url))
```
Comment (bryevdv) [inline sdk/identity/azure-identity/tests/helpers.py]: why not use `add_discrepancy` for this?

File: sdk/identity/azure-identity/tests/pod-identity/readme.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+Create the managed identity:
+```sh
+az identity create -g $RESOURCE_GROUP -n $MANAGED_IDENTITY_NAME
+```
+
+Save its `clientId`, `id` (ARM URI), and `principalId` (object ID) for later:
+```sh
+export MANAGED_IDENTITY_CLIENT_ID=$(az identity show -g $RESOURCE_GROUP -n $MANAGED_IDENTITY_NAME --query clientId -o tsv) \
+       MANAGED_IDENTITY_ID=$(az identity show -g $RESOURCE_GROUP -n $MANAGED_IDENTITY_NAME --query id -o tsv) \
+       MANAGED_IDENTITY_PRINCIPAL_ID=$(az identity show -g $RESOURCE_GROUP -n $MANAGED_IDENTITY_NAME --query principalId -o tsv)
```
Comment (bryevdv) [inline sdk/identity/azure-identity/tests/pod-identity/readme.md]: nit: (also, opinion) maybe three `export` lines with no line continuations?

**Comment by chlowell** [comment]
I moved the AKS bits to #9495.

**Pull request merged**